Add the audittrail adapter

linki opened the pull request:
Adds the audittrail adapter

Commit:
chore: add the audittrail adapter

Commit:
chore: add the audittrail adapter

Commit:
fix: add cluster id to audittrail webhook

Commit:
fix: add cluster id to audittrail webhook

Raffo committed:
Added configuration for the audit webhook

Raffo commented:
I added some of the config as taken from the kubernetes repo which contains some interesting observation: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L734

Raffo committed:
Modified audittrail-adapter to be a daemonset.

szuecs commented:
@Raffo why this should be a daemonset?

mikkeloscar committed:
Use kubernetes.io/role label for nodeAffinity

Uses the more "official" kubernetes.io/role label rather than our custom
master=true label.

Raffo commented:
@szuecs Because we decided not to have the adapter in the apiserver pod and we need the apiserver container to know the adapter's URL. Originally in this PR it was a deployment with a service, but this doesn't work as our apiserver is using `hostnetwork: true`. In this way, I solved the issue by changing it to a daemonset (running only on master nodes) and have each apiserver talk only to the local adapter. Feel free to change it if you want to implement it in a different way.

Raffo commented on cluster/manifests/audittrail-adapter/daemonset.yaml:
```diff
@@ -25,8 +25,10 @@ spec:
           requiredDuringSchedulingIgnoredDuringExecution:
             nodeSelectorTerms:
             - matchExpressions:
-              - key: master
```
Just note that this is also what we use in the skipper daemonset, I did it for consistency, maybe you can change it there as well (https://github.com/zalando-incubator/kubernetes-on-aws/blob/7acbcb75a1e30cba5c51254954fd231774a40b45/cluster/manifests/skipper/daemonset.yaml#L33 )

mikkeloscar replied:
Done #776

mikkeloscar committed:
Add Master node toleration

mikkeloscar committed:
Use Request level as default for audit events

linki committed:
chore: update audittrail-adapter to master-8

mikkeloscar committed:
Add credentials set

Commit:
fix: don't audit list events

Commit:
chore: use latest version of audittrail adapter

Commit:
fix: don't audit read-only events

mikkeloscar committed:
Open audittrail-adapter metrics port to worker nodes

linki committed:
chore: update audittrail to master-11

mikkeloscar committed:
Don't audit events from kube-system serviceaccounts

mikkeloscar committed:
Don't audit kubelet events

mikkeloscar committed:
Don't audit events from kube-controller-manager

mikkeloscar committed:
Correctly disable audit events from kube-system serviceaccounts

mikkeloscar committed:
Disable audit events from system:unsecured

mikkeloscar committed:
Disable audit events from system:apiserver user

mikkeloscar committed:
Correctly disable events from system:serviceaccounts:kube-system

mikkeloscar committed:
Update audittrail image

mikkeloscar committed:
Only run audittrail-adapter in production clusters

mikkeloscar commented:
:+1:

szuecs commented:
:+1:
